src/main/java/de/php_perfect/intellij/ddev/error_reporting/SentrySubmissionResult.java

```diff
@@ -0,0 +1,29 @@
+package de.php_perfect.intellij.ddev.error_reporting;
+
+import io.sentry.hints.SubmissionResult;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.function.Consumer;
+
+public class SentrySubmissionResult implements SubmissionResult {
+    private Consumer<Boolean> onResult;
+
+    public SentrySubmissionResult(@NotNull Consumer<Boolean> onResult) {
+        this.onResult = onResult;
+    }
+
+    @Override
+    public void setResult(boolean success) {
+        if (this.onResult == null) {
+            throw new IllegalStateException("result has already been set once");
+        }
+
+        this.onResult.accept(success);
+        this.onResult = null;
+    }
+
+    @Override
+    public boolean isSuccess() {
+        throw new UnsupportedOperationException();
```

Comment:
Is this a good idea?

Reply:
> The result itself is never accessed or used by the error reported.
> That's why I decided to throw an `UnsupportedOperationException` and not store the result somewhere.
> It is given to the Consumer once and after that the result object is basically useless as the consumer gets set to null too.
7e826d703cfd8da229f526c9de6e5f28b244240f

Would that explanation and making the class package private, to ensure it is not misused in some other place, be enough or would you prefer the interface being properly implemented?

Reply:
Not sure. I'm just not sure about the consequences. Implementing an interface means fulfilling a contract. Throwing such an exception seems like an ugly violation of the contract.
I think I would like to avoid this SubmissionResult at all, as it feels like abusing the interface. Probably this is something that might make us problems problems in future if there are any API changes.